websites/Y/Yume Nikki Online Project/presence.ts

```diff
@@ -0,0 +1,30 @@
+const presence = new Presence({
+		clientId: "1060269808383430696",
+	}),
+	elapsed = Math.floor(Date.now() / 1000),
+	strings = presence.getStrings({
+		play: "presence.playback.playing",
+		pause: "presence.playback.paused",
+	});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "yume",
+		startTimestamp: elapsed,
+	};
+
+	if (location.pathname === "/" || location.href.split(location.host)[1].toLowerCase() === "/index") {
+		presenceData.details = "The Nexus";
+		presenceData.state = "(Selecting Game)";
+
+	} else if (document.title.split(" - ")[0].toString() != "YNOproject") {
+		presenceData.details = document.title.split(" - ")[0].toString();
+		if (document.getElementById("locationText").innerText) {presenceData.state = document.getElementById("locationText").innerText;}
+		else {presenceData.state = "In Menu"}
+
+	} else {
+		presenceData.details = "Loading..."
+	}
+
+	presence.setActivity(presenceData);
+});
```

Comment:
Try to support more pages/features. Atm its a low quality presence so it the PR could get closed.

Reply:
Not sure what other features you would need, there's only the homepage and the games on the website, every other link just goes to external social media pages. The games themselves are walking simulators, meaning they are exclusively based on exploration and atmospheric experience, so the location of the player is the most important thing anyway.
Also, I am very sorry this seems to be too messy, I am new to this type of thing and don't really know what I'm doing. I made a presence for Terminal 00 a long time ago that was even simpler than this one, and it seems a lot has changed with the process for submitting these since then...